inc/class-rest-api.php

```diff
@@ -1586,4 +1587,22 @@ function ( $matches ) {
 
 		return null;
 	}
+
+	/**
+	 * Force a 255-character limit on the GUID of incoming attachment data.
+	 *
+	 * Intended for the {@see 'wp_insert_attachment_data'} filter.
+	 *
+	 * @see https://core.trac.wordpress.org/ticket/47296,
+	 *      https://core.trac.wordpress.org/ticket/32315.
+	 *
+	 * @param array $data Sanitized attachment post data.
+	 * @return array Updated post data.
+	 */
+	public function filter_attachment_guid( $data ) {
+		if ( ( isset( $data['post_type'] ) && 'attachment' === $data['post_type'] ) && isset( $data['guid'] ) ) {
```

Comment:
What's the reason for
`( isset( $data['post_type'] ) && 'attachment' === $data['post_type'] )`
over something like
`'attachment' === ( $data['post_type'] ?? '' )` ?

Personal preference? Compatibility with older versions of PHP? Performance?

Reply:
either is fine. I chose this because I copied it from somewhere else 😆